nat-lab/tests/test_mesh_exit_through_peer.py

```diff
@@ -91,8 +114,21 @@ async def test_mesh_exit_through_peer(
             )
         )
 
-        async with Ping(connection_alpha, beta.ip_addresses[0]).run() as ping:
-            await testing.wait_long(ping.wait_for_next_ping())
+        if exit_ip_stack in [IPStack.IPv4, IPStack.IPv4v6]:
+            async with Ping(
+                connection_alpha,
+                testing.unpack_optional(beta.get_ip_address(IPProto.IPv4)),
+                ip_proto=IPProto.IPv4,
+            ).run() as ping:
+                await testing.wait_long(ping.wait_for_next_ping())
+
+        if exit_ip_stack in [IPStack.IPv6, IPStack.IPv4v6]:
+            async with Ping(
```

Comment:
Could we adjust the `Ping` function to detect the IP version? This creates a lot of duplicate code.

Reply:
Done!

Reply:
:thinking: But it is still duplicated for IPv4 and IPv6. It would be nice in the test to not have two ping's. Maybe it would make sense to create some helper function or `utils.py` function which would take whatever arguments needed and do the pinging over both protocols internally?